src/io/hdf_stl.h

```diff
@@ -123,8 +123,6 @@ template<> struct h5data_proxy<std::string>
       }
       ref.resize(dim_out);
       herr_t ret = H5Dread(dataset, datatype, H5S_ALL, H5S_ALL, xfer_plist,&(ref[0]));
-      // Erase trailing null character
-      ref.erase (dim_out-1, 1);
```

Comment:
This line is not trailing the null character but the last one anyway.
When I hit here last time, I noticed that even if I left the last character, it was not the one written in the h5.
If you would like to remove these two lines, please check if the last character can be read correctly.